partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/TypePrecedence.kt

```diff
@@ -0,0 +1,39 @@
+package org.partiql.spi.function.builtins
+
+import org.partiql.types.PType.Kind
+
+/**
+ * @return the precedence of the types for the PartiQL comparator.
+ * @see .TYPE_PRECEDENCE
+ */
+@Suppress("deprecation")
+internal val TYPE_PRECEDENCE: Map<Kind, Int> = listOf(
```

Comment:
Top-level interval vals are ok afaik (functions were the leaked apis), but wouldn't hurt to put in a TypePrecedence object as a static field, kotlin is doing that anyways so let's at least control it.